mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -134,6 +134,8 @@ createGenISAFpToFp(GENX::FpToFpOp op, llvm::IRBuilderBase &builder,
   // TODO: Add verifier.
   assert(srcTySizeInBits != resTySizeInBits &&
          "Expecting first argument and result size to be different");
+  assert((op.getRoundingMode() || srcTySizeInBits < resTySizeInBits) &&
```

Comment:
We need to enforce this constraints in the verification code for this operation. It can be done in a different PR.